tests/harness.ps1

```
@@ -398,6 +398,30 @@ $hiCollectAppend = $AllText -match [regex]::Escape('set "HP_PYI_HID_COLLECT=%HP_
+# -- a reintroduced per-call reset would silently wipe an earlier call's own accumulated flags
+# on the second pass. An exact occurrence count of the bare reset line catches this: DLLBIND is
+# reset twice on purpose (the fresh-build-attempt reset plus :run_entry_smoke's own end-of-pass
+# trailer, pre-existing); the hidden-import pair is reset once (the fresh-build-attempt reset
+# only -- no end-of-pass trailer reset for these two, since HP_PYI_DLLBIND's own precedent never
+# needed one either).
+$dllbindResetCount   = ([regex]::Matches($AllText, [regex]::Escape('set "HP_PYI_DLLBIND="'))).Count
+$hiddenImportsResetCount = ([regex]::Matches($AllText, [regex]::Escape('set "HP_PYI_HIDDEN_IMPORTS="'))).Count
+$hidCollectResetCount    = ([regex]::Matches($AllText, [regex]::Escape('set "HP_PYI_HID_COLLECT="'))).Count
+$noPerCallResetLeak = ($dllbindResetCount -eq 2) -and ($hiddenImportsResetCount -eq 1) -and ($hidCollectResetCount -eq 1)
+$hasDllBundleSecondPass = $dllSecondCall -and $hiSecondCall -and $dllRebuildHasHid -and $dllRepairedSet -and $dllRepairedCheck -and $noPerCallResetLeak
+Write-Result 'batch.dll_bundle.second_pass' 'CLAUDE.md Item 29: a second :dll_bundle_recover pass runs after :hidden_import_recover, threading its own accumulated hidden-import flags through the DLL rebuild, then giving :hidden_import_recover one more bounded pass if a DLL was actually bundled -- with a regression guard on the exact-once-per-fresh-build-attempt reset counts for the three cross-call accumulator variables' $hasDllBundleSecondPass @{ dllSecondCall=$dllSecondCall; hiSecondCall=$hiSecondCall; dllRebuildHasHid=$dllRebuildHasHid; dllRepairedSet=$dllRepairedSet; dllRepairedCheck=$dllRepairedCheck; dllbindResetCount=$dllbindResetCount; hiddenImportsResetCount=$hiddenImportsResetCount; hidCollectResetCount=$hidCollectResetCount }
```

PSScriptAnalyzer (1.25.0)

[info] Cmdlet 'Write-Result' has positional parameter. Please use named parameters instead of positional parameters when calling a command.

(PSAvoidUsingPositionalParameters)